libraries/AP_SerialManager/AP_SerialManager.cpp

```diff
@@ -569,7 +569,13 @@ AP_HAL::UARTDriver *AP_SerialManager::find_serial(enum SerialProtocol protocol,
         return nullptr;
     }
     const uint8_t serial_idx = _state - &state[0];
-    return hal.serial(serial_idx);
+
+    // set options before any user does begin()
+    AP_HAL::UARTDriver *port = hal.serial(serial_idx);
+    if (port) {
+        port->set_options(_state->options);
```

Comment:
wouldn't it be better to have this in begin method itself, basically UART driver fetches the options?

Reply:
set_options() is also used to changes options from places other than the parameters, so if we always fetched from the parameters we'd get the wrong value. See calls like this:
  libraries/AP_RCProtocol/AP_RCProtocol.cpp:            added.uart->set_options(added.uart->get_options() | AP_HAL::UARTDriver::OPTION_RXINV);